### PR TITLE
En 4077 refactor shard processor constructor

### PR DESCRIPTION
**Pull request by miiu96**
Refactor shard processor constructor.
All constructor parameters was moved in a new struct.
Modify failing tests.

### CODE REVIEW

File: process/block/argsProcessor.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package block
```
Comment (iulianpascalau) [inline process/block/argsProcessor.go]: As to be consistent, should rename the file and structs to argProcessor (no s)

File: process/block/baseProcess_test.go [hunk 2]
```diff
@@ -314,24 +314,38 @@ func (wr *wrongBody) IsInterfaceNil() bool {
 	return false
 }
 
+func CreateMockArguments() blproc.ArgsShardProcessor {
+	arguments := blproc.ArgsShardProcessor{
+		ArgsBaseProcessor: &blproc.ArgsBaseProcessor{
```
Comment (iulianpascalau) [inline process/block/baseProcess_test.go]: Should instantiate the fields with the most used variants as seen in shardBlock_test.go

File: cmd/node/factory/structs.go [hunk 3]
```diff
@@ -1503,21 +1503,25 @@ func newShardBlockProcessorAndTracker(
...
+		RequestHandler:   requestHandler,
+		Core:             coreServiceContainer,
+	}
+	arguments := block.ArgsShardProcessor{
+		ArgsBaseProcessor: &argumentsBaseProcessor,
+		DataPool:          data.Datapool,
+		BlocksTracker:     blockTracker,
+		TxCoordinator:     txCoordinator,
+	}
+	blockProcessor, err := block.NewShardProcessor(arguments)
```
Comment (sasurobert) [inline cmd/node/factory/structs.go]: empty line before

File: process/block/argsProcessor.go (hunk 1 above)
Comment (miiu96) [reply process/block/argsProcessor.go]: Done.

File: cmd/node/factory/structs.go (hunk 3 above)
Comment (miiu96) [reply cmd/node/factory/structs.go]: Done.

File: process/block/baseProcess_test.go (hunk 2 above)
Comment (miiu96) [reply process/block/baseProcess_test.go]: Done.

**Pull request merged**